storage/watchable_store.go

```diff
@@ -240,7 +236,7 @@ func (s *watchableStore) syncWatchers() {
 		limit := cap(w.ch) - len(w.ch)
 		// the channel is full, try it in the next round
 		if limit == 0 {
-			nws = append(nws, w)
+			nws[w] = struct{}{}
```

Comment:
I mean there is no need to reconstruct the whole map. you could just delete unwanted ones from the original map `s.unsynced`.